public/lib/dragAndDroplayercontrol/__tests__/layerControlTree.js

```diff
@@ -0,0 +1,252 @@
+    const component = getMountedComponent();
+    expect(fakeEsClient.search.called);
+    expect(component.getElement('ul.euiTreeView'));
+    done();
+  });
+
+
+  it('should populate correct stored layers list with given aggs.' +
+   'This implicitly tests _calculateAllTypeCounts() and _getParent() and _getItems()', () => {
+
+    const expectedStoredLayersList = [{
+      label: 'Universe',
+      id: 'universe',
+      checked: true,
+      filtered: false,
+      group: true,
+      count: 34,
+      isParentItem: false,
+      path: 'universe',
+      itemInGroupChecked: true,
+      hasLayerSelect: true,
+      children: [
+        {
+          label: 'Countries',
+          id: 'universe/countries',
+          checked: true,
+          filtered: false,
+          children: [],
+          group: false,
+          count: 12,
+          isParentItem: false,
+          path: 'universe/countries'
+        },
+        {
+          label: 'Planets',
+          id: 'universe/planets',
+          checked: true,
+          filtered: false,
+          children: [
+            {
+              label: 'Planets',
+              id: 'alluniverse/planets',
+              checked: true,
+              filtered: false,
+              children: [],
+              group: false,
+              count: 10,
+              isParentItem: true,
+              path: 'universe/planets'
+            },
+            {
+              label: 'Countries',
+              id: 'universe/planets/countries',
+              checked: true,
+              filtered: false,
+              children: [],
+              group: false,
+              count: 12,
+              isParentItem: false,
+              path: 'universe/planets/countries'
+            }
+          ],
+          group: true,
+          count: 22,
+          isParentItem: false,
+          path: 'universe/planets',
+          itemInGroupChecked: true,
+          hasLayerSelect: true
+        }
+      ]
+    }];
+
+
+
+
+
+
+    function compareItems(item, fakeItem) {
+      expect(item.checked).to.be(fakeItem.checked);
+      expect(item.count).to.be(fakeItem.count);
+      expect(item.filtered).to.be(fakeItem.filtered);
+      expect(item.id).to.be(fakeItem.id);
+      expect(item.label).to.be(fakeItem.label);
+      expect(item.path).to.be(fakeItem.path);
+      expect(item.isParentItem).to.be(fakeItem.isParentItem);
+      expect(item.group).to.be(fakeItem.group);
+      expect(item.hasLayerSelect).to.be(fakeItem.hasLayerSelect);
+      if (item.group) {
+        expect(item.children.length).to.be(fakeItem.children.length);
+        for (let i = 0; i < item.children.length; i++) {
+          compareItems(item.children[i], fakeItem.children[i]);
+        }
+      }
+    }
+    const componentInstance = getMountedComponent().instance();
+    const storedLayersList = componentInstance._makeUiTreeStructure(fakeAggs);
+    compareItems(storedLayersList[0], expectedStoredLayersList[0]);
```

Comment:
can you try 
```suggestion
    expect(storedLayersList[0]).to.eql(expectedStoredLayersList[0]);
```
and if it works just remove the compareItems function